### PR TITLE
Enable NRT annotations in `SongSelect` classes

### PR DESCRIPTION
**Pull request by mk56-spn**
title.

### CODE REVIEW

File: osu.Game/Screens/Select/SongSelect.cs [hunk 1]
```diff
@@ -85,50 +82,49 @@ public abstract partial class SongSelect : ScreenWithBeatmapBackground, IKeyBind
         public bool BeatmapSetsLoaded => IsLoaded && Carousel?.BeatmapSetsLoaded == true;
 
         [Resolved]
-        private Bindable<IReadOnlyList<Mod>> selectedMods { get; set; }
+        private Bindable<IReadOnlyList<Mod>> selectedMods { get; set; } = null!;
 
-        protected BeatmapCarousel Carousel { get; private set; }
+        protected BeatmapCarousel? Carousel { get; private set; }
```
Comment (frenzibyte) [inline osu.Game/Screens/Select/SongSelect.cs:87]: Not sure how to feel about marking beatmap carousel as nullable, it's guaranteed not to be null anytime after loading the screen. We usually suppress null in that case similar to other components initialized on BDL and resolved dependencies.

That being said, special consideration needs to be taken on any usage of the screen before load (such as `BeatmapSetsLoaded`, potentially)

**Review by frenzibyte — CHANGES_REQUESTED** [review]

File: osu.Game/Screens/Select/SongSelect.cs (hunk 1 above)
Comment (mk56-spn) [reply osu.Game/Screens/Select/SongSelect.cs:87]: yeah this was irking me ( the fact that it was guaranteed not to be null most of the  time) but i didn't know how to address it without making it nullable, ill go change it

File: osu.Game/Screens/Select/SongSelect.cs [hunk 2]
```diff
@@ -854,28 +855,31 @@ private void bindBindables()
         /// Will immediately run filter operations if required.
         /// </summary>
         /// <returns>Whether a transfer occurred.</returns>
-        private bool transferRulesetValue()
+        private bool transferRulesetValue(bool carouselLoaded = true)
```
Comment (peppy) [inline osu.Game/Screens/Select/SongSelect.cs:855]: This is bad and cannot exist

Comment (mk56-spn) [reply osu.Game/Screens/Select/SongSelect.cs:855]: Wha would be a better way of blocking usage without making the carousel nullable?

Comment (peppy) [reply osu.Game/Screens/Select/SongSelect.cs]: Check my better solution.

**Review by peppy — APPROVED** [review]

**Pull request merged**